### PR TITLE
update weights (0.9.33)

### PR DESCRIPTION
**Pull request by coderobe**

### CODE REVIEW

File: runtime/kusama/src/weights/frame_election_provider_support.rs [hunk 1]
```diff
@@ -48,20 +48,22 @@ impl<T: frame_system::Config> frame_election_provider_support::WeightInfo for We
 	/// The range of component `t` is `[500, 1000]`.
 	/// The range of component `d` is `[5, 16]`.
 	fn phragmen(v: u32, _t: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 56_000
-			.saturating_add(Weight::from_ref_time(13_944_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 4_876_000
-			.saturating_add(Weight::from_ref_time(2_223_649_000 as u64).saturating_mul(d as u64))
+		// Minimum execution time: 5_474_633 nanoseconds.
+		Weight::from_ref_time(5_525_752_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/frame_election_provider_support.rs:52]: big jump up from 0

File: runtime/kusama/src/weights/frame_election_provider_support.rs [hunk 2]
```diff
@@ -48,20 +48,22 @@ impl<T: frame_system::Config> frame_election_provider_support::WeightInfo for We
...
 	/// The range of component `t` is `[500, 1000]`.
 	/// The range of component `d` is `[5, 16]`.
 	fn phragmms(v: u32, _t: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 79_000
-			.saturating_add(Weight::from_ref_time(14_480_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 6_844_000
-			.saturating_add(Weight::from_ref_time(2_525_332_000 as u64).saturating_mul(d as u64))
+		// Minimum execution time: 4_309_239 nanoseconds.
+		Weight::from_ref_time(4_353_424_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/frame_election_provider_support.rs:63]: big jump up from 0

File: runtime/kusama/src/weights/pallet_conviction_voting.rs [hunk 3]
```diff
@@ -49,34 +48,38 @@ impl<T: frame_system::Config> pallet_conviction_voting::WeightInfo for WeightInf
...
 	// Storage: ConvictionVoting ClassLocksFor (r:1 w:1)
 	// Storage: Balances Locks (r:1 w:1)
-	// Storage: Scheduler Agenda (r:2 w:2)
+	// Storage: Scheduler Agenda (r:1 w:1)
 	fn vote_new() -> Weight {
-		Weight::from_ref_time(5_652_326_000 as u64)
-			.saturating_add(T::DbWeight::get().reads(6 as u64))
-			.saturating_add(T::DbWeight::get().writes(6 as u64))
+		// Minimum execution time: 129_807 nanoseconds.
+		Weight::from_ref_time(133_007_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_conviction_voting.rs:54]: big jump down

File: runtime/kusama/src/weights/pallet_election_provider_multi_phase.rs [hunk 4]
```diff
@@ -139,16 +145,11 @@ impl<T: frame_system::Config> pallet_election_provider_multi_phase::WeightInfo f
...
-			.saturating_add(Weight::from_ref_time(874_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 14_000
-			.saturating_add(Weight::from_ref_time(39_000 as u64).saturating_mul(t as u64))
-			// Standard Error: 23_000
-			.saturating_add(Weight::from_ref_time(10_873_000 as u64).saturating_mul(a as u64))
-			// Standard Error: 35_000
-			.saturating_add(Weight::from_ref_time(2_257_000 as u64).saturating_mul(d as u64))
+	fn submit_unsigned(_v: u32, _t: u32, a: u32, _d: u32, ) -> Weight {
+		// Minimum execution time: 6_866_951 nanoseconds.
+		Weight::from_ref_time(6_994_679_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_election_provider_multi_phase.rs:150]: big jump up from 0

File: runtime/kusama/src/weights/pallet_election_provider_multi_phase.rs [hunk 5]
```diff
@@ -160,16 +161,13 @@ impl<T: frame_system::Config> pallet_election_provider_multi_phase::WeightInfo f
...
-			.saturating_add(Weight::from_ref_time(895_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 25_000
-			.saturating_add(Weight::from_ref_time(18_000 as u64).saturating_mul(t as u64))
-			// Standard Error: 42_000
-			.saturating_add(Weight::from_ref_time(8_673_000 as u64).saturating_mul(a as u64))
-			// Standard Error: 63_000
-			.saturating_add(Weight::from_ref_time(1_598_000 as u64).saturating_mul(d as u64))
+	fn feasibility_check(v: u32, _t: u32, a: u32, _d: u32, ) -> Weight {
+		// Minimum execution time: 5_710_244 nanoseconds.
+		Weight::from_ref_time(5_739_503_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_election_provider_multi_phase.rs:166]: big jump up from 0

File: runtime/kusama/src/weights/pallet_elections_phragmen.rs [hunk 6]
```diff
@@ -148,13 +158,12 @@ impl<T: frame_system::Config> pallet_elections_phragmen::WeightInfo for WeightIn
...
-	fn clean_defunct_voters(v: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 88_000
-			.saturating_add(Weight::from_ref_time(60_894_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 88_000
-			.saturating_add(Weight::from_ref_time(379_000 as u64).saturating_mul(d as u64))
+	/// The range of component `d` is `[0, 5000]`.
+	fn clean_defunct_voters(v: u32, _d: u32, ) -> Weight {
+		// Minimum execution time: 279_914_581 nanoseconds.
+		Weight::from_ref_time(280_093_180_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_elections_phragmen.rs:164]: huge jump up from 0

File: runtime/kusama/src/weights/pallet_elections_phragmen.rs [hunk 7]
```diff
@@ -172,14 +181,16 @@ impl<T: frame_system::Config> pallet_elections_phragmen::WeightInfo for WeightIn
 	/// The range of component `v` is `[1, 10000]`.
 	/// The range of component `e` is `[10000, 160000]`.
 	fn election_phragmen(c: u32, v: u32, e: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 691_000
-			.saturating_add(Weight::from_ref_time(57_805_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 46_000
-			.saturating_add(Weight::from_ref_time(3_139_000 as u64).saturating_mul(e as u64))
+		// Minimum execution time: 29_009_063 nanoseconds.
+		Weight::from_ref_time(29_156_039_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_elections_phragmen.rs:185]: big jump up from 0

File: runtime/kusama/src/weights/pallet_staking.rs [hunk 8]
```diff
@@ -365,27 +386,28 @@ impl<T: frame_system::Config> pallet_staking::WeightInfo for WeightInfo<T> {
...
 	// Storage: Staking Validators (r:501 w:0)
 	// Storage: System BlockWeight (r:1 w:1)
 	/// The range of component `v` is `[500, 1000]`.
 	fn get_npos_targets(v: u32, ) -> Weight {
-		Weight::from_ref_time(310_326_000 as u64)
-			// Standard Error: 56_000
-			.saturating_add(Weight::from_ref_time(6_164_000 as u64).saturating_mul(v as u64))
-			.saturating_add(T::DbWeight::get().reads(2 as u64))
+		// Minimum execution time: 3_520_733 nanoseconds.
+		Weight::from_ref_time(3_602_387_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/kusama/src/weights/pallet_staking.rs:407]: x10 up

File: runtime/polkadot/src/weights/frame_election_provider_support.rs [hunk 9]
```diff
@@ -48,20 +48,22 @@ impl<T: frame_system::Config> frame_election_provider_support::WeightInfo for We
 	/// The range of component `t` is `[500, 1000]`.
 	/// The range of component `d` is `[5, 16]`.
 	fn phragmen(v: u32, _t: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 56_000
-			.saturating_add(Weight::from_ref_time(13_900_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 4_843_000
-			.saturating_add(Weight::from_ref_time(2_216_812_000 as u64).saturating_mul(d as u64))
+		// Minimum execution time: 5_475_839 nanoseconds.
+		Weight::from_ref_time(5_538_536_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/frame_election_provider_support.rs:52]: big jump from 0

File: runtime/polkadot/src/weights/frame_election_provider_support.rs [hunk 10]
```diff
@@ -48,20 +48,22 @@ impl<T: frame_system::Config> frame_election_provider_support::WeightInfo for We
...
 	/// The range of component `t` is `[500, 1000]`.
 	/// The range of component `d` is `[5, 16]`.
 	fn phragmms(v: u32, _t: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 84_000
-			.saturating_add(Weight::from_ref_time(14_955_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 7_259_000
-			.saturating_add(Weight::from_ref_time(2_615_111_000 as u64).saturating_mul(d as u64))
+		// Minimum execution time: 4_279_269 nanoseconds.
+		Weight::from_ref_time(4_313_963_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/frame_election_provider_support.rs:63]: big jump from 0

File: runtime/polkadot/src/weights/pallet_election_provider_multi_phase.rs [hunk 11]
```diff
@@ -137,16 +145,13 @@ impl<T: frame_system::Config> pallet_election_provider_multi_phase::WeightInfo f
...
-			.saturating_add(Weight::from_ref_time(765_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 7_000
-			.saturating_add(Weight::from_ref_time(34_000 as u64).saturating_mul(t as u64))
-			// Standard Error: 12_000
-			.saturating_add(Weight::from_ref_time(7_231_000 as u64).saturating_mul(a as u64))
-			// Standard Error: 18_000
-			.saturating_add(Weight::from_ref_time(1_703_000 as u64).saturating_mul(d as u64))
+	fn submit_unsigned(v: u32, _t: u32, a: u32, _d: u32, ) -> Weight {
+		// Minimum execution time: 4_693_663 nanoseconds.
+		Weight::from_ref_time(4_728_052_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/pallet_election_provider_multi_phase.rs:150]: big jump up from 0

File: runtime/polkadot/src/weights/pallet_election_provider_multi_phase.rs [hunk 12]
```diff
@@ -158,16 +163,13 @@ impl<T: frame_system::Config> pallet_election_provider_multi_phase::WeightInfo f
...
-			.saturating_add(Weight::from_ref_time(789_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 8_000
-			.saturating_add(Weight::from_ref_time(41_000 as u64).saturating_mul(t as u64))
-			// Standard Error: 13_000
-			.saturating_add(Weight::from_ref_time(5_713_000 as u64).saturating_mul(a as u64))
-			// Standard Error: 20_000
-			.saturating_add(Weight::from_ref_time(1_311_000 as u64).saturating_mul(d as u64))
+	fn feasibility_check(v: u32, _t: u32, a: u32, _d: u32, ) -> Weight {
+		// Minimum execution time: 3_895_772 nanoseconds.
+		Weight::from_ref_time(3_953_808_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/pallet_election_provider_multi_phase.rs:168]: big jump up from 0

File: runtime/polkadot/src/weights/pallet_elections_phragmen.rs [hunk 13]
```diff
@@ -148,13 +158,12 @@ impl<T: frame_system::Config> pallet_elections_phragmen::WeightInfo for WeightIn
...
-	fn clean_defunct_voters(v: u32, d: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 86_000
-			.saturating_add(Weight::from_ref_time(60_379_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 86_000
-			.saturating_add(Weight::from_ref_time(543_000 as u64).saturating_mul(d as u64))
+	/// The range of component `d` is `[0, 5000]`.
+	fn clean_defunct_voters(v: u32, _d: u32, ) -> Weight {
+		// Minimum execution time: 281_848_917 nanoseconds.
+		Weight::from_ref_time(282_529_259_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/pallet_elections_phragmen.rs:164]: huge jump up from 0

File: runtime/polkadot/src/weights/pallet_elections_phragmen.rs [hunk 14]
```diff
@@ -172,14 +181,16 @@ impl<T: frame_system::Config> pallet_elections_phragmen::WeightInfo for WeightIn
 	/// The range of component `v` is `[1, 10000]`.
 	/// The range of component `e` is `[10000, 160000]`.
 	fn election_phragmen(c: u32, v: u32, e: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 674_000
-			.saturating_add(Weight::from_ref_time(52_473_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 44_000
-			.saturating_add(Weight::from_ref_time(2_826_000 as u64).saturating_mul(e as u64))
+		// Minimum execution time: 28_514_414 nanoseconds.
+		Weight::from_ref_time(28_653_120_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/pallet_elections_phragmen.rs:185]: big jump from 0

File: runtime/polkadot/src/weights/pallet_fast_unstake.rs [hunk 15]
```diff
@@ -50,65 +49,72 @@ impl<T: frame_system::Config> pallet_fast_unstake::WeightInfo for WeightInfo<T>
...
 	// Storage: Staking CurrentEra (r:1 w:0)
 	// Storage: Staking ErasStakers (r:56 w:0)
 	/// The range of component `x` is `[28, 3584]`.
 	fn on_idle_check(x: u32, ) -> Weight {
-		Weight::from_ref_time(504_317_000 as u64)
-			// Standard Error: 18_023
-			.saturating_add(Weight::from_ref_time(14_203_535 as u64).saturating_mul(x as u64))
-			.saturating_add(T::DbWeight::get().reads(64 as u64))
+		// Minimum execution time: 28_585_887 nanoseconds.
+		Weight::from_ref_time(28_897_826_000 as u64)
```
Comment (EgorPopelyaev) [inline runtime/polkadot/src/weights/pallet_fast_unstake.rs:77]: x40 jump up

File: runtime/rococo/src/weights/pallet_elections_phragmen.rs [hunk 16]
```diff
@@ -170,14 +181,16 @@ impl<T: frame_system::Config> pallet_elections_phragmen::WeightInfo for WeightIn
...
 	/// The range of component `e` is `[10000, 160000]`.
 	fn election_phragmen(c: u32, v: u32, e: u32, ) -> Weight {
-		Weight::from_ref_time(0 as u64)
-			// Standard Error: 1_864_000
-			.saturating_add(Weight::from_ref_time(167_385_000 as u64).saturating_mul(v as u64))
-			// Standard Error: 124_000
-			.saturating_add(Weight::from_ref_time(9_721_000 as u64).saturating_mul(e as u64))
+		// Minimum execution time: 28_205_490 nanoseconds.
+		Weight::from_ref_time(28_372_658_000 as u64)
+			// Standard Error: 537_591
```
Comment (EgorPopelyaev) [inline runtime/rococo/src/weights/pallet_elections_phragmen.rs:186]: big jump up from 0

**Comment by coderobe** [comment]
swc https://weights.tasty.limo/compare?unit=weight&ignore_errors=true&threshold=10&method=guess-worst&repo=polkadot&old=release-v0.9.32&new=coderobe/weights-33&path_pattern=runtime/*/src/weights/**/*.rs,bridges/modules/*/src/weights.rs

**Comment by coderobe** [comment]
is this fine? cc @ggwpez @kianenigma 
image

**Comment by ggwpez** [comment]
The idle checks should be fine with any weight, since they only happen when nothing else is scheduled and can limit their consumption to the remaining weight of the block.

PS: @coderobe your SWC link above has `release-v0.9.32` instead of 33. There is https://weights.tasty.limo/compare-mr for easier selection so it becomes https://weights.tasty.limo/compare?repo=polkadot&threshold=10&path_pattern=runtime%2F*%2Fsrc%2Fweights%2F**%2F*.rs%2Cbridges%2Fmodules%2F*%2Fsrc%2Fweights.rs&method=guess-worst&ignore_errors=true&unit=time&old=release-v0.9.33&new=coderobe/weights-33

**Review by ggwpez — APPROVED** [review]
Given the changes in Substrate I dont see anything *too weird*.

**Comment by coderobe** [comment]
> PS: @coderobe your SWC link above has `release-v0.9.32` instead of 33.

@ggwpez i'm intentionally comparing with the previous release weights to xref with the weights we're on now, because i think they might not have been backported to master/this branch

**Review by koute — APPROVED** [review]
For what's worth, looks good to me.

The big changes from zero base weight to non-zero base weight for `pallet_elections_phragmen`/`pallet_staking`/`frame_election_provider_support`/`pallet_election_provider_multi_phase` are expected. Would be [nice to improve those](https://github.com/paritytech/substrate/issues/12652) but even as-is the change in the final minimums/maximums are within an acceptable range.

Another big bump (over a million percent) to the minimums is with the parachain code upgrades, and looking at the code that seems reasonable to me (if anything, to me it looks like the minimums before were horribly wrong, which is to be expected as that benchmark triggered the zero base weight bug).

One weight could probably use its bounds extended to cover zero, for which [I made a PR here](https://github.com/paritytech/polkadot/pull/6318). Not a big deal though, and in practice won't make much of a difference.

The one huge jump I can't really explain is the `on_idle_check` from `pallet_fast_unstake`, but the benchmark code looks reasonable; did it just suddenly got more expensive? Would be nice for someone who's familiar with that code to double-check that as both the minimum and the maximum increased by 5000%, and the number of DB writes also drastically increased.

**Pull request merged**